File: app/src/main/java/com/chooloo/www/callmanager/ui/contact/ContactContract.java
package com.chooloo.www.callmanager.ui.contact;

import android.app.Activity;

import com.chooloo.www.callmanager.database.entity.Contact;
import com.chooloo.www.callmanager.ui.base.BaseContract;

public class ContactContract {
    public interface View extends BaseContract.View {
        Contact getContact();

        Activity getActivity();

        void setUp();

        void toggleFavIcon(boolean toggle);

        void setContactFromIntent();

        void checkNoRecents();

        void onBackPressed();
    }

    public interface Presenter<V extends View> extends BaseContract.Presenter<V> {
        void setActivity();

        void onBackButtonPressed();

        void onActionCall();

        void onActionSms();

        void onActionEdit();

        void onActionInfo();

        void onActionDelete();

        void onActionFav();
    }
}

File: chooloolib/src/main/java/com/chooloo/www/chooloolib/ui/contact/ContactContract.kt
package com.chooloo.www.chooloolib.ui.contact

import com.chooloo.www.chooloolib.ui.base.BaseContract

interface ContactContract : BaseContract {
    interface View : BaseContract.View {

    }

    interface Controller<V : View> : BaseContract.Controller<V> {

    }
}